runtime/jvmti/jvmtiHelpers.c

```diff
@@ -174,20 +174,25 @@ disposeEnvironment(J9JVMTIEnv * j9env, UDATA freeData)
 #if JAVA_SPEC_VERSION >= 11
 		if (j9env->capabilities.can_generate_sampled_object_alloc_events) {
 			J9JVMTI_DATA_FROM_VM(vm)->flags &= ~J9JVMTI_FLAG_SAMPLED_OBJECT_ALLOC_ENABLED;
+			/* Set sampling interval to UDATA_MAX to inform GC that sampling is not required */
+			vm->memoryManagerFunctions->j9gc_set_allocation_sampling_interval(vm, UDATA_MAX);
 		}
 #endif /* JAVA_SPEC_VERSION >= 11 */
 
 		/* Remove all breakpoints */
 
-		if (j9env->breakpoints != NULL) {
+		if (NULL != j9env->breakpoints) {
 			J9VMThread * currentThread = vm->internalVMFunctions->currentVMThread(vm);
-			pool_state poolState;
-			J9JVMTIAgentBreakpoint * agentBreakpoint = NULL;
-
-			agentBreakpoint = pool_startDo(j9env->breakpoints, &poolState);
-			while (agentBreakpoint != NULL) {
-				deleteAgentBreakpoint(currentThread, j9env, agentBreakpoint);
-				agentBreakpoint = pool_nextDo(&poolState);
+			/* Ignore the case that currentThread is NULL, i.e., unattached thread */
```

Comment:
Why is this necessary?

Reply:
`currentThread` is from `vm->internalVMFunctions->currentVMThread(vm)` which might return `NULL` as per https://github.com/eclipse/openj9/blob/5800af9452097b029c1a38fbc5cbd20011060a8b/runtime/jvmti/jvmtiHelpers.c#L502-L510
Current function `disposeEnvironment()` doesn't return an error code, hence just call `deleteAgentBreakpoint(currentThread, j9env, agentBreakpoint)` when the `currentThread` is not`NULL`, and ignore the `NULL` case.

Reply:
Is this to fix something? I don't believe this situation could actually occur, so I'd rather see the crash than a silent no-op.

Reply:
No, this change is completely from eyeballing the code. Will restore the code then.